Add guidelines for how to use GPG with Git. (#45)
abtion/guidelines: gpg-signing/README.md

# GPG Signing with Git
## Issues
### Commits cannot be signed
Run `export GPG_TTY=$(tty)` in your terminal.

### Commits are not signed automatically
To set up GPG signed commits, first run:

```sh
git config --local commit.gpgSign true
git config --local user.signingKey YOUR_GPG_KEY
```

You may want to use `--global` instead if you want to set up GPG signing for all
your projects (which is required for Abtion projects).

> To avoid having to sign all of your commits, you can create a

Reviewer: Do you mean

```suggestion
To avoid having to enter you password for each commit, you can create a
```

Author: Good catch. :-)

`~/.gnupg/gpg-agent.conf` file with the following content:

```sh
default-cache-ttl 3600
max-cache-ttl 3600
```

> This ensures that your commits are automatically signed for one hour, and after
> one hour you need to provide your GPG password again. Run `gpgconf --reload
> gpg-agent` to reload your updated GPG configuration.

Reviewer: ```suggestion
This way you only have to provide your password once every hour.
Run `gpgconf --reload gpg-agent` to reload your updated GPG configuration.
```
